src/TraceEvent/TraceEvent.cs

```
@@ -3046,6 +3046,7 @@ private void DoDispatch(TraceEvent anEvent)
             {
                 Debug.WriteLine("Error: exception thrown during callback.  Will be swallowed!");
                 Debug.WriteLine("Exception: " + e.Message);
+                Debug.Assert(false, "Thrown exception " + e.GetType().Name +  " '" + e.Message + "'");
```

Here is the exception handler that was catching exceptions and swallowing them.